tests/unit/rpc/handlers/DepositAuthorizedTests.cpp

```diff
@@ -531,3 +553,285 @@ TEST_F(RPCDepositAuthorizedTest, DifferentAccountsWithDepositAuthFlagReturnsTrue
         EXPECT_EQ(*output.result, json::parse(expectedOut));
     });
 }
+
+TEST_F(RPCDepositAuthorizedTest, CredentialAcceptedAndNotExpiredReturnsTrue)
+{
+    static auto const expectedOut = fmt::format(
+        R"({{
+            "ledger_hash": "4BC50C9B0D8515D3EAAE1E74B29A95804346C491EE1A95BF25E4AAB854A6A652",
+            "ledger_index": 30,
+            "validated": true,
+            "deposit_authorized": true,
+            "source_account": "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
+            "destination_account": "rLEsXccBGNR3UPuPu2hUXPjziKC3qKSBun",
+            "credentials": ["{}"]
+        }})",
+        CREDENTIALHASH  // CREDENTIALHASH should match credentialIndex
+    );
+
+    backend->setRange(10, 30);
+
+    auto ledgerHeader = CreateLedgerHeader(LEDGERHASH, 30);
+
+    ON_CALL(*backend, fetchLedgerByHash(ripple::uint256{LEDGERHASH}, _)).WillByDefault(Return(ledgerHeader));
+    EXPECT_CALL(*backend, fetchLedgerByHash).Times(1);
```

Comment:
`Times(1)` is by default. You don't need to explicitly specify it.

Use `EXPECT_CALL(...).WillOnce(...)` instead of two lines (here and below).